_posts/2020-07-15-#19031.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: pr
+date: 2020-07-15
+title: "Implement ADDRv2 support (part of BIP155)"
+pr: 19031
+authors: [vasild]
+components: ["p2p"]
+host: troygiorshev
+status: upcoming
+commit:
+---
+
+## Notes
+
+* [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is the implementation of [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki), the `addrv2` message.  PR 19031 is being used as a roadmap, where [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360) is currently ready for review.
+
+* The `addr` message is currently used to relay IP addresses across the network.  It uses only IPv6 addresses, where IPv4 addresses are given as IPv4-mapped IPv6 addresses.  More information about it can be found [here](https://btcinformation.org/en/developer-reference#addr).  The `addrv2` message will allow for longer variable length addresses.
+
+* Bitcoin currently has Tor integration.  Relevant to this PR is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address.  Ultimately these changes will allow us to work with [Tor v3](https://github.com/bitcoin/bitcoin/issues/18884), along with many other different types of networks.
```

Comment:
suggest: work with Tor v3 ([meta issue here](https://github.com/bitcoin/bitcoin/issues/18884))


I think the notes ought to describe Tor v3 or link to a good resource. One resource I found was: https://trac.torproject.org/projects/tor/wiki/doc/NextGenOnions; perhaps you know a better one.